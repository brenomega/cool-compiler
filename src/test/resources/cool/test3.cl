-- Arquivo de teste com 2 classes

class Pessoa{
    nome : String <- "";

    set_nome(n : String) : Pessoa{
        nome <- n; self
    };

    get_nome() : String{
        nome
    };
};

class Main inherits IO {
    p1 : Pessoa <- new Pessoa;
    p2 : Pessoa <- new Pessoa;

    main() : Object {
        (* Inicializando os nomes*)
        p1 <- p1.set_nome("André");
        p2 <- p2.set_nome("Carlos");

        (* Escrevendo os nomes *)
        out_string(p1.get_nome());
        out_string(" é amigo de ");
        out_string(p2.get_nome());
        out_string("\n");

        0
    };

};

