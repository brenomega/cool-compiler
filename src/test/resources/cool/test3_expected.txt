Line 1: COMMENT('-- Arquivo de teste com 2 classes')
Line 3: WHITESPACE('

')
Line 3: CLASS('class')
Line 3: WHITESPACE(' ')
Line 3: TYPEID('Pessoa')
Line 3: LBRACE('{')
Line 4: WHITESPACE('
    ')
Line 4: OBJECTID('nome')
Line 4: WHITESPACE(' ')
Line 4: COLON(':')
Line 4: WHITESPACE(' ')
Line 4: TYPEID('String')
Line 4: WHITESPACE(' ')
Line 4: ASSIGN('<-')
Line 4: WHITESPACE(' ')
Line 4: STRING('')
Line 4: SEMI(';')
Line 6: WHITESPACE('

    ')
Line 6: OBJECTID('set_nome')
Line 6: LPAREN('(')
Line 6: OBJECTID('n')
Line 6: WHITESPACE(' ')
Line 6: COLON(':')
Line 6: WHITESPACE(' ')
Line 6: TYPEID('String')
Line 6: RPAREN(')')
Line 6: WHITESPACE(' ')
Line 6: COLON(':')
Line 6: WHITESPACE(' ')
Line 6: TYPEID('Pessoa')
Line 6: LBRACE('{')
Line 7: WHITESPACE('
        ')
Line 7: OBJECTID('nome')
Line 7: WHITESPACE(' ')
Line 7: ASSIGN('<-')
Line 7: WHITESPACE(' ')
Line 7: OBJECTID('n')
Line 7: SEMI(';')
Line 7: WHITESPACE(' ')
Line 7: OBJECTID('self')
Line 8: WHITESPACE('
    ')
Line 8: RBRACE('}')
Line 8: SEMI(';')
Line 10: WHITESPACE('

    ')
Line 10: OBJECTID('get_nome')
Line 10: LPAREN('(')
Line 10: RPAREN(')')
Line 10: WHITESPACE(' ')
Line 10: COLON(':')
Line 10: WHITESPACE(' ')
Line 10: TYPEID('String')
Line 10: LBRACE('{')
Line 11: WHITESPACE('
        ')
Line 11: OBJECTID('nome')
Line 12: WHITESPACE('
    ')
Line 12: RBRACE('}')
Line 12: SEMI(';')
Line 13: WHITESPACE('
')
Line 13: RBRACE('}')
Line 13: SEMI(';')
Line 15: WHITESPACE('

')
Line 15: CLASS('class')
Line 15: WHITESPACE(' ')
Line 15: TYPEID('Main')
Line 15: WHITESPACE(' ')
Line 15: INHERITS('inherits')
Line 15: WHITESPACE(' ')
Line 15: TYPEID('IO')
Line 15: WHITESPACE(' ')
Line 15: LBRACE('{')
Line 16: WHITESPACE('
    ')
Line 16: OBJECTID('p1')
Line 16: WHITESPACE(' ')
Line 16: COLON(':')
Line 16: WHITESPACE(' ')
Line 16: TYPEID('Pessoa')
Line 16: WHITESPACE(' ')
Line 16: ASSIGN('<-')
Line 16: WHITESPACE(' ')
Line 16: NEW('new')
Line 16: WHITESPACE(' ')
Line 16: TYPEID('Pessoa')
Line 16: SEMI(';')
Line 17: WHITESPACE('
    ')
Line 17: OBJECTID('p2')
Line 17: WHITESPACE(' ')
Line 17: COLON(':')
Line 17: WHITESPACE(' ')
Line 17: TYPEID('Pessoa')
Line 17: WHITESPACE(' ')
Line 17: ASSIGN('<-')
Line 17: WHITESPACE(' ')
Line 17: NEW('new')
Line 17: WHITESPACE(' ')
Line 17: TYPEID('Pessoa')
Line 17: SEMI(';')
Line 19: WHITESPACE('

    ')
Line 19: OBJECTID('main')
Line 19: LPAREN('(')
Line 19: RPAREN(')')
Line 19: WHITESPACE(' ')
Line 19: COLON(':')
Line 19: WHITESPACE(' ')
Line 19: TYPEID('Object')
Line 19: WHITESPACE(' ')
Line 19: LBRACE('{')
Line 20: WHITESPACE('
        ')
Line 20: COMMENT('(* Inicializando os nomes*)')
Line 21: WHITESPACE('
        ')
Line 21: OBJECTID('p1')
Line 21: WHITESPACE(' ')
Line 21: ASSIGN('<-')
Line 21: WHITESPACE(' ')
Line 21: OBJECTID('p1')
Line 21: DOT('.')
Line 21: OBJECTID('set_nome')
Line 21: LPAREN('(')
Line 21: STRING('André')
Line 21: RPAREN(')')
Line 21: SEMI(';')
Line 22: WHITESPACE('
        ')
Line 22: OBJECTID('p2')
Line 22: WHITESPACE(' ')
Line 22: ASSIGN('<-')
Line 22: WHITESPACE(' ')
Line 22: OBJECTID('p2')
Line 22: DOT('.')
Line 22: OBJECTID('set_nome')
Line 22: LPAREN('(')
Line 22: STRING('Carlos')
Line 22: RPAREN(')')
Line 22: SEMI(';')
Line 24: WHITESPACE('

        ')
Line 24: COMMENT('(* Escrevendo os nomes *)')
Line 25: WHITESPACE('
        ')
Line 25: OBJECTID('out_string')
Line 25: LPAREN('(')
Line 25: OBJECTID('p1')
Line 25: DOT('.')
Line 25: OBJECTID('get_nome')
Line 25: LPAREN('(')
Line 25: RPAREN(')')
Line 25: RPAREN(')')
Line 25: SEMI(';')
Line 26: WHITESPACE('
        ')
Line 26: OBJECTID('out_string')
Line 26: LPAREN('(')
Line 26: STRING(' é amigo de ')
Line 26: RPAREN(')')
Line 26: SEMI(';')
Line 27: WHITESPACE('
        ')
Line 27: OBJECTID('out_string')
Line 27: LPAREN('(')
Line 27: OBJECTID('p2')
Line 27: DOT('.')
Line 27: OBJECTID('get_nome')
Line 27: LPAREN('(')
Line 27: RPAREN(')')
Line 27: RPAREN(')')
Line 27: SEMI(';')
Line 28: WHITESPACE('
        ')
Line 28: OBJECTID('out_string')
Line 28: LPAREN('(')
Line 28: STRING('
')
Line 28: RPAREN(')')
Line 28: SEMI(';')
Line 30: WHITESPACE('

        ')
Line 30: INT('0')
Line 31: WHITESPACE('
    ')
Line 31: RBRACE('}')
Line 31: SEMI(';')
Line 33: WHITESPACE('

')
Line 33: RBRACE('}')
Line 33: SEMI(';')
Line 35: WHITESPACE('

')
Line 35: EOF('')